frontend/src/pages/Task.tsx

```diff
@@ -0,0 +1,630 @@
+                                buttonIcon="edit"
+                                onAction={() => setIsEditingTaskPrompt(true)}
+                            />
+                        )}
+                    </div>
+                </div>
+
+                <div className="flex flex-col w-full gap-2">
+                    <textarea
+                        className=" bg-dark-200 disabled:bg-transparent focus:border-primary-100 border-gray-200/60 focus:ring-0 focus:shadow-none rounded-sm h-56 text-gray-400"
+                        disabled={!isEditingTaskPrompt}
+                        value={taskPrompt}
+                        placeholder="Enter prompt..."
+                        onChange={(evt) => setTaskPrompt(evt.target.value)}
+                    />
+                </div>
+            </div>
+        </Card>
+    );
+}
+
+function TaskValidation({ task }: { task: TaskInfo }) {
+    // State Management for Component
+    const isActive = task.output_grammar ? true : false;
+    const [validationActive, setValidationActive] = useState<boolean>(isActive);
+    const [editingGrammer, setEditingGrammar] = useState<boolean>(!isActive);
+    const [taskGrammar, setTaskGrammar] = useState<string>(
+        `${task.output_grammar?.grammar_user_code ? task.output_grammar?.grammar_user_code : ""}`
+    );
+
+    // Query and Mutation Hooks
+    const [updateGrammarModelAction] = useUpdateGrammarModeMutation();
+
+    // Event Handlers
+    const onDownload = () => {
+        if (!task.output_grammar?.grammar_generated) {
+            console.log("Error: No generated grammar was loaded from the server.");
+            return;
+        }
+        const file = new File([task.output_grammar?.grammar_generated], "grammar.gbnf", {
+            type: "text/plain",
+        });
+
+        const link = document.createElement("a");
+        const url = URL.createObjectURL(file);
+
+        link.href = url;
+        link.download = file.name;
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+        window.URL.revokeObjectURL(url);
+    };
+
+    const onGenerate = () => {
+        try {
+            const srcFile = ts.createSourceFile("source.ts", taskGrammar, ts.ScriptTarget.ESNext);
+            const ifaces: Array<string> = [];
+            srcFile.forEachChild((child) => {
+                if (ts.isInterfaceDeclaration(child)) {
+                    ifaces.push(child.name.getText(srcFile));
+                }
+            });
+            const grammarArray = compile(taskGrammar, ifaces[0]);
+            const grammarFile = serializeGrammar(grammarArray);
+
+            updateGrammarModelAction({
+                task: task.name,
+                grammar: {
+                    grammar_user_code: taskGrammar,
+                    grammar_generated: grammarFile,
+                },
+            });
+
+            setEditingGrammar(false);
+        } catch (e) {
```

Comment:
What in here actually throws an exception (vs e.g. the RTK query populating the `error` value)